service/custom_logs_post.go

```diff
@@ -2,53 +2,40 @@ package service
 
 import (
 	"encoding/json"
-	"fmt"
 	"net/http"
 
 	"github.com/bitrise-io/api-utils/httpresponse"
-	"go.uber.org/zap"
+	"github.com/bitrise-io/bitrise-step-analytics/models"
+	"github.com/pkg/errors"
 )
 
-const (
-	logLvlInfo  string = "info"
-	logLvlWarn  string = "warn"
-	logLvlError string = "error"
-)
-
-// LogPostRequestBody ...
-type LogPostRequestBody struct {
-	LogLevel string                 `json:"log_level"`
-	Message  string                 `json:"message"`
-	Data     map[string]interface{} `json:"data"`
-}
-
 // CustomLogsPostHandler ...
 func CustomLogsPostHandler(w http.ResponseWriter, r *http.Request) error {
-	logData := LogPostRequestBody{}
+	var log models.RemoteLog
 	defer httpresponse.RequestBodyCloseWithErrorLog(r)
-	if err := json.NewDecoder(r.Body).Decode(&logData); err != nil {
+	if err := json.NewDecoder(r.Body).Decode(&log); err != nil {
 		return httpresponse.RespondWithBadRequestError(w, "Invalid request body, JSON decode failed")
 	}
 
-	logger, err := zap.NewProduction()
-	if err != nil {
-		fmt.Printf("Failed to initialize logger: %s", err)
+	if len(log.LogLevel) == 0 {
+		return httpresponse.RespondWithBadRequestError(w, "Invalid request body, please provide log_level")
+	}
+	if len(log.Message) == 0 {
+		return httpresponse.RespondWithBadRequestError(w, "Invalid request body, please provide message")
+	}
+	if _, ok := log.Data["step_id"].(string); !ok {
```

Comment:
NOTE: by having a function for getting a field from data, some duplication could be avoided:

```
func (a RemoteLog) DataField(key string) (string, bool) {
	d, ok := a.data[key].(string)
	return d, ok
}
```

also could be used here: https://github.com/bitrise-io/bitrise-step-analytics/pull/13/files#diff-ff296a0cb815ef5bd00483b0178b2d4dR21